src/transformers/modeling_utils.py

```diff
@@ -729,6 +733,13 @@ def generate(
         else:
             assert input_ids.dim() == 2, "Input prompt should be of shape (batch_size, sequence length)."
 
+        if pad_token_id is None:
```

Comment:
I think setting special tokens to -1 wouldn't be a good idea:
In the case of batch_size > 1 if one sentence stops at a eos_token_id it would be padded by -1 tokens afterwards and still be used as an input for the model.forward fn if other sentences in the batch haven't finished yet.
This would create an error. See also PR #2635 and #2885